src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -158,21 +156,21 @@ private void TryKickOffNextBatchTask()
                 // No in-flight task.  Kick one off to process these messages a second from now.
                 // We always attach the task to the previous one so that notifications to the ui
                 // follow the same order as the notification the OOP server sent to us.
-                var token = _asyncListener.BeginAsyncOperation(nameof(TryKickOffNextBatchTask));
+                _updateTask = ContinueAfterDelay();
+                _taskInFlight = true;
```

Comment:
❗ If `ContinueAfterDelay` completes synchronously, this line will latch `_taskInFlight` to true and no further work will be processed.

Reply:
good catch.  fixed.